core/App/ehsm_napi.cpp

```diff
@@ -272,41 +315,145 @@ char* NAPI_Decrypt(const char* cmk,
     plaint_data.datalen = 0;
     ret = Decrypt(&masterkey, &cipher_data, &aad_data, &plaint_data);
     if (ret != EH_OK) {
-        retJsonObj.code = retJsonObj.CODE_FAILED;
-        retJsonObj.msg = "Failed to get size of Decrypt";
+        retJsonObj.setCode(retJsonObj.CODE_FAILED);
+        retJsonObj.setMessage("Server exception.");
         goto out;
     }
 
     plaint_data.data = (uint8_t*)malloc(plaint_data.datalen);
     if (plaint_data.data == NULL) {
-        retJsonObj.code = retJsonObj.CODE_FAILED;
-        retJsonObj.msg = "Failed to alloc memory";
+        retJsonObj.setCode(retJsonObj.CODE_FAILED);
+        retJsonObj.setMessage("Server exception.");
         goto out;
     }
 
     ret = Decrypt(&masterkey, &cipher_data, &aad_data, &plaint_data);
     if (ret != EH_OK) {
-        retJsonObj.code = retJsonObj.CODE_FAILED;
-        retJsonObj.msg = "Failed to decrypt data";
+        retJsonObj.setCode(retJsonObj.CODE_FAILED);
+        retJsonObj.setMessage("Server exception.");
         goto out;
     }
 
     plaintext_base64 = base64_encode(plaint_data.data, plaint_data.datalen);
     if(plaintext_base64.size() > 0){
-        /*todo: make the unittest case happy, remove it later after added the c++ json parser*/
+        retJsonObj.addData("plaintext_base64", plaintext_base64);
         SAFE_FREE(masterkey.keyblob);
         SAFE_FREE(plaint_data.data);
-        return StringToChar(plaintext_base64);
-        //retJsonObj.code = retJsonObj.CODE_SUCCESS;
-        //retJsonObj.addData("key", plaintext_base64);
+        Finalize();
+        return retJsonObj.toChar();
     }
 out:
     SAFE_FREE(masterkey.keyblob);
     SAFE_FREE(plaint_data.data);
+    Finalize();
+    return retJsonObj.toChar();
+}
 
+/*
+@return
+[string] json string
+    {
+        code: int,
+        message: string,
+        result: {
+            plaintext_base64 : string,
+            ciphertext_base64 : string,
+        }
+    }
+*/
+char* NAPI_GenerateDataKey(const char* cmk_base64,
+        const char* aad)
+{
```

Comment:
seems this API are defined wrong!!
please see the definitions in the header file:
https://github.com/intel/ehsm/blob/main/core/App/ehsm_napi.h#L90
char* NAPI_GenerateDataKey(const char* cmk,
        const uint32_t keylen,
        const char* aad);

Reply:
you have missed the argument "keylen"...

Reply:
It will be fix.